sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -132,41 +158,6 @@ def _validate_parameters(self):
 
         The parameters that are directly passed to the grower are checked in
         TreeGrower."""
-
-        if self.loss not in self._VALID_LOSSES and not isinstance(self.loss, BaseLoss):
-            raise ValueError(
-                "Loss {} is not supported for {}. Accepted losses: {}.".format(
-                    self.loss, self.__class__.__name__, ", ".join(self._VALID_LOSSES)
-                )
-            )
-
-        if self.learning_rate <= 0:
-            raise ValueError(
-                "learning_rate={} must be strictly positive".format(self.learning_rate)
-            )
-        if self.max_iter < 1:
-            raise ValueError(
-                "max_iter={} must not be smaller than 1.".format(self.max_iter)
-            )
-        if self.n_iter_no_change < 0:
-            raise ValueError(
-                "n_iter_no_change={} must be positive.".format(self.n_iter_no_change)
-            )
-        if self.validation_fraction is not None and self.validation_fraction <= 0:
-            raise ValueError(
-                "validation_fraction={} must be strictly positive, or None.".format(
-                    self.validation_fraction
-                )
-            )
-        if self.tol < 0:
-            raise ValueError("tol={} must not be smaller than 0.".format(self.tol))
-
-        if not (2 <= self.max_bins <= 255):
-            raise ValueError(
-                "max_bins={} should be no smaller than 2 "
-                "and no larger than 255.".format(self.max_bins)
-            )
-
         if self.monotonic_cst is not None and self.n_trees_per_iteration_ != 1:
             raise ValueError(
                 "monotonic constraints are not supported for multiclass classification."
```

Comment:
Could this also be integrated into `_parameter_constraints`? If so, we can remove `validate_parameters` function and only use `self._validate_params`

Reply:
I think it's a bit too specific to this estimator to be included in `_parameter_constraints`.

Reply:
constraints that depend on the value of another parameter cannot be set in `_parameter_constraints`

Reply:
From my experience, libraries that add constraints for two or more parameters will end up similar to [ConfigSpace's API](https://automl.github.io/ConfigSpace/master/API-Doc.html#conditions). In the end, I think adding constraints conditioned on two parameters would be too complex and make it harder to contribute.

Reply:
Thanks for the comments! I think we can keep this check in this PR since it is too complex for `_parameter_constraints`.